### PR TITLE
Add required field indicator text

### LINKED ISSUE #2629: (WCAG 2.1) Required fields should be identified on first page load.
Feedback from a11y experts during an a11y audit.

Severity (Critical/Serious/Moderate/Good to have): Serious

Required fields should be identified on first page load. We should identify required fields both visually and programmatically. 

Visually, we could: (1) put a "required" text label, or (2) mark it with a star. Programmatically, we should mark the html as required, either with (1) html required attribute, or (2) aria-required attribute. The html attribute is preferred, because it has baked in information.

### PR DESCRIPTION
**Pull request by arielwainer**
### Description

Indicate requiredness with a * and helper text based on similar change made in https://github.com/civiform/civiform/pull/3168
image

## Release notes:

Add indicators for which questions are required when filling out a form.

### Checklist

- [ ] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)

Fixes #2629

### CODE REVIEW

File: server/app/views/applicant/ApplicantProgramBlockEditView.java [hunk 1]
```diff
@@ -113,6 +115,7 @@ private ContainerTag<?> renderBlockWithSubmitForm(Params params) {
         .withAction(formAction)
         .withMethod(HttpVerbs.POST)
         .with(makeCsrfTokenInputTag(params.request()))
+        .with(requiredFieldsExplanationContent(params.messages()))
```
Comment (jeanniefu) [inline server/app/views/applicant/ApplicantProgramBlockEditView.java:118]: Ah I noticed the file screenshot is missing this text. Can you also edit the file upload logic (line 101)?

**Review by jeanniefu — COMMENTED** [review]
nit: Update PR description

**Comment by shanemc-goog** [comment]
Please fill out the PR template: description, Release Notes and the checkboxes as relevant

File: server/app/views/applicant/ApplicantProgramBlockEditView.java (hunk 1 above)
Comment (arielwainer) [reply server/app/views/applicant/ApplicantProgramBlockEditView.java:118]: Added a duplicate method in fileUploadViewStrategy but it feels redundant to have identical methods twice. What would be the best shared location for requiredFieldsExplanationContent()?

Comment (jeanniefu) [reply server/app/views/applicant/ApplicantProgramBlockEditView.java:118]: Seems reasonable to me to put it in the parent class (ApplicationBaseView)

File: server/app/views/questiontypes/ApplicantQuestionRendererImpl.java [hunk 2]
```diff
@@ -98,7 +98,8 @@ public final DivTag render(ApplicantQuestionRendererParams params) {
 
     ImmutableList<DomContent> questionTextDoms =
         TextFormatter.createLinksAndEscapeText(
-            question.getQuestionText(), TextFormatter.UrlOpenAction.NewTab);
+            !question.isOptional() ? question.getQuestionText() + "*" : question.getQuestionText(),
```
Comment (shanemc-goog) [inline server/app/views/questiontypes/ApplicantQuestionRendererImpl.java:101]: Suggestion: swap the order so the assertion is positive which aids readability.  isOptional ?   :

**Review by shanemc-goog — COMMENTED** [review]
Looks good!!! Thank you!

**Review by shanemc-goog — APPROVED** [review]

**Review by jeanniefu — APPROVED** [review]
Thanks so much for this change!

**Pull request merged**